### PR TITLE
chore: An attempt at a simpler bug report / pull request form

### PR DESCRIPTION
**Pull request by threepointone**
This is an attempt at simplifying our bug report / pull request forms. Some of it might seem controversial, so approach it with an open mind. Let's have a conversation here!

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-template.yaml [hunk 1]
```diff
@@ -1,86 +1,28 @@
...
+title: "\U0001F41B Bug:"
 labels: ["bug"]
 
 body:
   - type: markdown
     attributes:
       value: |
-        ## Quick Bug Form
-        Thank you for taking the time to file a bug report! Please fill out this form as completely as possible.
-  - type: dropdown
```
Comment (threepointone) [inline .github/ISSUE_TEMPLATE/bug-template.yaml:12]: This is usually wrangler or c3, and when it's not, the user will tell us what the product is. We can make the form less intimidating by removing it.

File: .github/ISSUE_TEMPLATE/bug-template.yaml [hunk 2]
```diff
@@ -1,86 +1,28 @@
...
-        - KV Asset Handler
-        - Workers Vitest Integration
-        - Other
-    validations:
-      required: true
+        ## Please provide as many details as possible regarding your bug.
   - type: input
     attributes:
-      label: What version(s) of the tool(s) are you using?
-      placeholder: 0.0.0 [Wrangler], 0.0.0 [C3], etc.
```
Comment (threepointone) [inline .github/ISSUE_TEMPLATE/bug-template.yaml:38]: Let's have them add the node version here, and make it optional. We aim to support all (modern) versions, and if a bug is for an odler version it's usually immediately apparent.

File: .github/ISSUE_TEMPLATE/bug-template.yaml [hunk 3]
```diff
@@ -1,86 +1,28 @@
...
-      placeholder: 0.0.0
-    validations:
-      required: false
-  - type: input
-    attributes:
-      label: What operating system and version are you using?
+      label: What Operating System are you using?
       placeholder: Mac Sonoma 14.2.1, Windows 11, Linux (or WSL) Debian Bookworm, etc.
     validations:
-      required: true
```
Comment (threepointone) [inline .github/ISSUE_TEMPLATE/bug-template.yaml:52]: We can ask as a followup if it's not obvious from the report.

File: .github/ISSUE_TEMPLATE/bug-template.yaml [hunk 4]
```diff
@@ -1,86 +1,28 @@
...
+      required: false
   - type: textarea
     attributes:
       label: Describe the Bug
-      description: Please describe the bug and include steps to reproduce.
-      value: |
-        ### Observed behavior
```
Comment (threepointone) [inline .github/ISSUE_TEMPLATE/bug-template.yaml:58]: The issue is meant to be a starting point for a conversation, unless it's obvious from the report. We can simplify the wording here.

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -4,30 +4,12 @@ Fixes #[insert GH or internal issue number(s)].
...
+- [ ] E2E Tests (Use "e2e" label to run them on CI, or ask maintainer to run separately)
+- [ ] [Changeset](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets)
+- [ ] Docs
 
 <!--
 Have you read our [Contributing guide](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md)?
-In particular, for non-trivial changes, please always engage on the issue or create a discussion or feature request issue first before writing your code.
 -->
 
-<!--
```
Comment (threepointone) [inline .github/pull_request_template.md:29]: This has never been used. Let's remove it.

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -4,30 +4,12 @@ Fixes #[insert GH or internal issue number(s)].
...
-  - [ ] Cloudflare docs PR(s): <!--e.g. <https://github.com/cloudflare/cloudflare-docs/pull/>...-->
-  - [ ] Documentation not necessary because:
+- [ ] Tests
+- [ ] E2E Tests (Use "e2e" label to run them on CI, or ask maintainer to run separately)
+- [ ] [Changeset](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets)
+- [ ] Docs
 
 <!--
 Have you read our [Contributing guide](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md)?
-In particular, for non-trivial changes, please always engage on the issue or create a discussion or feature request issue first before writing your code.
```
Comment (threepointone) [inline .github/pull_request_template.md:26]: People (like me) might use the PR itsel f as an engagement point (and we should actually encourage it).

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -4,30 +4,12 @@ Fixes #[insert GH or internal issue number(s)].
...
-  - [ ] Changeset included
-  - [ ] Changeset not necessary because:
-- Public documentation
-  - [ ] TODO (before merge)
-  - [ ] Cloudflare docs PR(s): <!--e.g. <https://github.com/cloudflare/cloudflare-docs/pull/>...-->
-  - [ ] Documentation not necessary because:
+- [ ] Tests
+- [ ] E2E Tests (Use "e2e" label to run them on CI, or ask maintainer to run separately)
+- [ ] [Changeset](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets)
+- [ ] Docs
```
Comment (penalosa) [inline .github/pull_request_template.md:10]: What's the main motivation here? There's definitely some simplification in order with this template, but it seems like the main change here is removing the requirement to justify omitting tests/changeset/etc.... Does that present a huge burden to external contributors?

File: .github/pull_request_template.md (hunk 6 above)
Comment (penalosa) [reply .github/pull_request_template.md:26]: One thing I'd be slightly concerned about with this change is an external contributor opening up a PR for a feature we don't want to add for whatever reason, and then us having to close the PR after they've already done a bunch of work (as opposed to closing an issue after just a suggestion of a feature). There probably won't be loads of cases of that though, to be fair, and maybe that's an okay tradeoff?

File: .github/pull_request_template.md (hunk 7 above)
Comment (threepointone) [reply .github/pull_request_template.md:10]: For internal contributors, we can always have higher standards. Those don't have to be reflected on the form.

File: .github/pull_request_template.md (hunk 6 above)
Comment (threepointone) [reply .github/pull_request_template.md:26]: Yup, that's what I think.

File: .github/pull_request_template.md (hunk 7 above)
Comment (RamIdeas) [reply .github/pull_request_template.md:10]: How do we enforce the higher standards on internal contributions?

I'd love to encourage more external contributions but Wrangler is an active product and internal contributions are far more common

File: .github/ISSUE_TEMPLATE/bug-template.yaml (hunk 2 above)
Comment (RamIdeas) [reply .github/ISSUE_TEMPLATE/bug-template.yaml:38]: Good shout collapsing the versions question into less inputs 👍

File: .github/pull_request_template.md (hunk 7 above)
Comment (threepointone) [reply .github/pull_request_template.md:10]: I guess my point is all the existing checkboxes don't really enforce it valuably enough to outweigh the cons, and we can still write automation around a smaller set of checkboxes.

Comment (threepointone) [reply .github/pull_request_template.md:10]: But part of me also wants to make it possible for internal folks to do driveby fixes with the same freedom we give external contributors. We use it the most ourselves so I bet we'd like to quickly send fixes too.

File: .github/pull_request_template.md (hunk 6 above)
Comment (petebacondarwin) [reply .github/pull_request_template.md:26]: I suspect what would actually happen in practice, rather than the external contributor going and creating an issue for discussion first, they would spend ages making the PR, then after pushing it realise that they need an issue, go create the issue, and then we have to close both the PR and issue if it is something we didn't want in the first place.

Comment (threepointone) [reply .github/pull_request_template.md:26]: To clarify, that means you're in support of removing this line/requirement right? Because you're describing the situation as it exists right now?

(I also want to make sure we're _not_ optimising to prevent the rare scenario of an external contributor making a big PR without ever talking to us. Has it happened often?)

Comment (lrapoport-cf) [reply .github/pull_request_template.md:26]: for some context here, the reason this was added is for exactly the scenario @penalosa cited -- we were finding ourselves having to close PRs that implemented features that hadn't been discussed with the team, but which didn't make sense to include for various reasons. the goal is to save folks from doing work that won't be landable. 

i agree that at the point of someone seeing this it's likely too late to save them doing the work, but i think having some language to signal process can 1) help manage expectations if the PR does end up being closed, and 2) reduce potential future friction

(as a note, while this is specified in the contributing guide [here](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md?plain=1#L9), it can be easily missed)

File: .github/pull_request_template.md (hunk 7 above)
Comment (lrapoport-cf) [reply .github/pull_request_template.md:10]: definitely want to reduce friction where possible. however my concern here is that by removing the multiple choice, there is no signal to our team on whether a checkbox was intentionally left unchecked or just forgotten. and if intentionally left unchecked, whether it was valid to do so. while we can engage in the PR on these items to clarify, it takes more overall back and forth than if the information were provided up front.

Comment (lrapoport-cf) [reply .github/pull_request_template.md:10]: i'd also be curious to understand better the extent of the con for external folks. do you know of cases where people don't put up PRs because of the checkboxes?

**Comment by penalosa** [comment]
Thanks for this—we're going to close this PR for now because it's a bit stale, but we'll add some of the bug report template modifications as a follow-up PR
